File: src/main/java/me/nepnep/nepbot/message/command/commands/Suggestion.java
package me.nepnep.nepbot.message.command.commands;

import me.nepnep.nepbot.message.command.Category;
import me.nepnep.nepbot.message.command.ICommand;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.Permission;
import net.dv8tion.jda.api.entities.TextChannel;
import net.dv8tion.jda.api.entities.User;
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;

import java.util.List;

public class Suggestion implements ICommand {
    @Override
    public void execute(List<String> args, GuildMessageReceivedEvent event) {

        TextChannel channel = event.getChannel();

        if (channel.getName().equals("suggestions")) {
            if (args.size() >= 1) {
                String suggestion = String.join(" ", args.subList(0, args.size()));
                User author = event.getAuthor();
                EmbedBuilder builder = new EmbedBuilder();

                builder.setTitle(author.getAsTag());

                builder.setThumbnail(author.getAvatarUrl());

                builder.setDescription(suggestion);

                builder.setFooter("ID: " + author.getId());

                channel.sendMessage(builder.build()).queue(message -> {
                    message.addReaction("✅").queue();
                    message.addReaction("❌").queue();
                });
                event.getMessage().delete().queue();
            } else {
                channel.sendMessage("Invalid usage!").queue();
            }
        } else {
            channel.sendMessage("Channel not named `suggestions`!").queue();
        }
    }
    @Override
    public String getInvoke() {
        return "suggestion";
    }
    @Override
    public Permission requiredPermission() {
        return null;
    }
    @Override
    public Category getCategory() {
        return Category.GENERAL;
    }
    @Override
    public String getDescription() {
        return "Makes a suggestion: ;suggestion <String suggestion>";
    }
}

File: src/main/kotlin/me/nepnep/nepbot/message/command/commands/Suggestion.kt
package me.nepnep.nepbot.message.command.commands

import dev.minn.jda.ktx.messages.Embed
import me.nepnep.nepbot.message.command.AbstractCommand
import me.nepnep.nepbot.message.command.Category
import net.dv8tion.jda.api.Permission
import net.dv8tion.jda.api.entities.GuildMessageChannel
import net.dv8tion.jda.api.entities.emoji.Emoji
import net.dv8tion.jda.api.events.message.MessageReceivedEvent

class Suggestion : AbstractCommand(
    "suggestion",
    Category.GENERAL,
    "Makes a suggestion: ;suggestion <String suggestion>"
) {
    override suspend fun execute(args: List<String>, event: MessageReceivedEvent, channel: GuildMessageChannel) {
        if (channel.name != "suggestions") {
            channel.sendMessage("Channel not named `suggestions`!").queue()
            return
        }
        if (args.isEmpty()) {
            channel.sendMessage("Invalid usage!").queue()
            return
        }
        if (!event.guild.selfMember.hasPermission(channel, Permission.MESSAGE_MANAGE)) {
            channel.sendMessage("I do not have permission to delete messages in this channel").queue()
            return
        }

        val suggestion = args.subList(0, args.size).joinToString(" ")
        val author = event.author

        val embed = Embed { 
            title = author.asTag
            thumbnail = author.effectiveAvatarUrl
            description = suggestion
            footer("ID: " + author.id)
        }
        channel.sendMessageEmbeds(embed).queue {
            it.addReaction(Emoji.fromUnicode("✅")).queue()
            it.addReaction(Emoji.fromUnicode("❌")).queue()
        }
        event.message.delete().queue()
    }
}